vscode-lean4/test/suite/bootstrap/bootstrap.test.ts

```diff
@@ -17,19 +13,27 @@ suite('Lean4 Bootstrap Test Suite', () => {
         logger.log('=================== Install elan on demand ===================');
         void vscode.window.showInformationMessage('Running tests: ' + __dirname);
 
+        cleanTempFolder('elan');
+
         // this will wait up to 60 seconds to do full elan lean install, so test machines better
         // be able to do that.
         const lean = await initLean4Untitled('#eval Lean.versionString');
         const info = lean.exports.infoProvider;
+        const  expected = '4.0.0-nightly-';
         assert(info, 'No InfoProvider export');
 
         // give it a extra long timeout in case test machine is really slow.
         logger.log('Wait for elan install of Lean nightly build...')
-        await waitForActiveClientRunning(lean.exports.clientProvider, 500);
-        const client = assertActiveClient(lean.exports.clientProvider);
-        // wait 60 seconds, if nothing then kick it with a restart server command, and try that up to 5 times
-        // and if it times out at 300 seconds then waitForInfoviewHtml prints the contents of the InfoView so we can see what happened.
-		await waitForInfoviewHtml(info, '4.0.0-nightly-', 5, 60000, false, () => restartLeanServer(client));
+        await waitForActiveClient(lean.exports.clientProvider, 120);
+        await waitForActiveClientRunning(lean.exports.clientProvider, 300);
+
+        // if it times out at 300 seconds then waitForInfoviewHtml prints the contents of the InfoView so we can see what happened.
+		await waitForInfoviewHtml(info, expected, 10, 10000, true, async () => {
+            // 60 seconds elapsed, and infoview is not updating, try and re-edit
+            // the file to force the LSP to update.
+            await deleteAllText();
```

Comment:
I am not sure about this part. I feel like "reedit if the infoview is stuck" is a workaround that users may choose for an infoview-not-updating bug, but our tests should not tolerate such bugs.

Reply:
Sure I'll comment it out, if tests remain reliable we can leave it out, but we also need reliable tests, so we shall see, sounds good?

Reply:
If this breaks the tests, either something in the test's assumptions is wrong or there is another extension bug to fix.